SonarQube.Client/RoslynExporterAdapter/SonarLintConfigGenerator.cs

```diff
@@ -0,0 +1,108 @@
+﻿/*
+ * SonarScanner for MSBuild
+ * Copyright (C) 2016-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using SonarQube.Client.Models;
+
+// Logically equivalent to the SonarScanner for MSBuild class "RoslynSonarLint"
```

Comment:
This class should be functionally equivalent to the S4MSB version.
Unfortunately, the S4MSB implementation uses a `StringBuilder` to build the XML, so I didn't keep any of the implementation. Also, the S4MSB version doesn't have any tests, so I couldn't resuse those either.

Reply:
I see that the original class references "secured" properties, is that not relevant in our case?
https://github.com/SonarSource/sonar-scanner-msbuild/blob/9ccfdb648a0411014b29c7aee8e347aeab87ea71/src/SonarScanner.MSBuild.PreProcessor/Roslyn/RoslynSonarLint.cs#L46

Reply:
No, I don't they exist any more. The secured properties were for the C++ and VB net plugins when they were commerically licensed.